common/djangoapps/student/helpers.py

```diff
@@ -268,14 +268,22 @@ def get_next_url_for_login_page(request):
                 # Tries reversing the LMS dashboard if the url doesn't exist
                 redirect_to = reverse('dashboard')
 
+            if redirect_to_logistration:
+                redirect_to = settings.LMS_ROOT_URL + reverse('dashboard')
+
         elif settings.ROOT_URLCONF == 'cms.urls':
             redirect_to = reverse('home')
 
+            if redirect_to_logistration:
+                redirect_to = settings.CMS_BASE + reverse('home')
```

Comment:
Use already created `redirect_to` variable instead of calling the reverse again and use the `configuration_helpers` to get settings value. You might also need to prepend the protocol `HTTP` or `HTTPS`, see this for reference https://github.com/edx/edx-platform/blob/af6cab86c33314c9fb0faabc4dd3bd9ba5230d97/openedx/core/djangoapps/xblock/apps.py#L83-L89